controllers/fluentdconfig_controller.go

```diff
@@ -113,29 +113,54 @@ func (r *FluentdConfigReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, err
 	}
 
+	// Loop over all Fluentd CRs
 	for _, fd := range fluentdList.Items {
 		// Get the config selector in this fluentd instance
 		fdSelector, err := metav1.LabelSelectorAsSelector(&fd.Spec.FluentdCfgSelector)
 		if err != nil {
-			// Patch this fluentd instance if the selectors exsit errors
+			// Patch this fluentd instance if the selectors exit with errors
 			if err := r.PatchObjects(ctx, &fd, fluentdv1alpha1.InactiveState, err.Error()); err != nil {
 				return ctrl.Result{}, err
 			}
 
 			continue
 		}
-
-		// A secret loader supports LoadSecret method to parse the targeted secret.
+		// A secret loader supports LoadSecret method to parse the targeted secret
 		sl := plugins.NewSecretLoader(r.Client, fd.Namespace, r.Log)
 
 		// gpr acts as a global resource to store the related plugin resources
 		gpr := fluentdv1alpha1.NewGlobalPluginResources("main")
 
 		// Each cluster/namespace scope fluentd configs will generate their own filters/outputs plugins with their own cfgId/cfgLabel,
-		// and they will finally be combined into one fluentd config file.
+		// and they will finally be combined into one fluentd config file
 		gpr.CombineGlobalInputsPlugins(sl, fd.Spec.GlobalInputs)
 
-		// globalCfgLabels stores cfgLabels, the same cfg label is not allowed.
+		// Default Output and filter
+		// list all namespaced CRs
+		filters, outputs, err := r.ListNamespacedLevelResources(ctx, fd.Namespace, fd.Spec.DefaultFilterSelector, fd.Spec.DefaultOutputSelector)
+		if err != nil {
+			r.Log.Info("List namespace level resources failed", "config", "default", "err", err.Error())
+			return ctrl.Result{}, err
+		}
+		if len(filters) > 0 || len(outputs) > 0 {
+			// Combine the namespaced filter/output pluginstores in this fluentd config
+			cfgResouces, errs := gpr.PatchAndFilterNamespacedLevelResources(sl, fmt.Sprintf("%s-%s-%s", fd.Kind, fd.Namespace, fd.Name), filters, outputs)
+			if len(errs) > 0 {
+				r.Log.Info("Patch and filter namespace level resources failed", "config", "default", "err", strings.Join(errs, ","))
+				return ctrl.Result{}, fmt.Errorf(strings.Join(errs, ","))
+			}
+
+			// WithCfgResources will collect all plugins to generate main config
+			err = gpr.WithCfgResources("@default", cfgResouces)
+			if err != nil {
+				r.Log.Info("Combine resources failed", "config", "default", "err", err.Error())
+				return ctrl.Result{}, err
+			}
+			// Add the default route to the main routing plugin
+			gpr.MainRouterPlugins.InsertPairs("default_route", "@default")
+		}
+
```

Comment:
@benjaminhuo @wenchajun this code block handles the the newly defined default selectors.

Reply:
Got you. Sorry, my bad it's folded by default.
Thanks anyway for this great enhancement! @Jakob3xD